### PR TITLE
Add class notice to the IDC container

### PR DESCRIPTION
**Pull request by htdat**
👋 Hi, I am creating this PR is more for a starting point for discussion. Also see p9dueE-3Xf-p2#comment-7005 and p1642030666001300/1639517545.011100-slack-CPBTESZMG

Related https://github.com/Automattic/woocommerce-payments/pull/3580

#### Changes proposed in this Pull Request:

When working in the WCPay PR above, I noticed that the IDC banner does not display in WooCommerce settings page such as: 

- wp-admin/admin.php?page=wc-settings
- wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments

That's because WooCommerce core (via its component woocommerce-admin) tries to stop admin notices tinkering before the WP core moves them below the page title _intentionally_ - [src](https://github.com/woocommerce/woocommerce-admin/blob/0b68c1d7c0c5025789071022efa129126da1ded5/src/Loader.php#L724-L739).

This PR is to allow the IDC banner for those pages by adding the `notice` class so that it can display under the admin page title. 

Note: the IDC banner does not display in the React-powered settings page too, and we should not try to display them that's because WooCommerce suppresses them all. See the src above.

- wp-admin/admin.php?page=wc-admin
- wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview

#### Jetpack product discussion

The downside of this approach is that the IDC banner may not really on the top of each admin page. 

#### Testing instructions:

* See the before and after rebuilding the Jetpack IDC package. 

Markup on 2022-01-09 at 16:24:47

### CODE REVIEW

File: projects/packages/identity-crisis/src/_inc/style.scss [hunk 1]
```diff
@@ -2,3 +2,8 @@
 	margin-top: 40px;
 	margin-bottom: 40px;
 }
+
+#jp-identity-crisis-container.notice {
+  background: none;
+  border: none;
```
Comment (jeherve) [inline projects/packages/identity-crisis/src/_inc/style.scss:8]: Tabs > spaces :)

```suggestion
	background: none;
	border: none;
```

**Review by jeherve — CHANGES_REQUESTED** [review]

**Comment by jeherve** [comment]
I have not tested this myself, so I'm adding the "Needs Team Review" label so someone from your team can take a look. 👍

File: projects/packages/identity-crisis/src/_inc/style.scss (hunk 1 above)
Comment (htdat) [reply projects/packages/identity-crisis/src/_inc/style.scss]: Thank you 👍

**Comment by htdat** [comment]
> I have not tested this myself, so I'm adding the "Needs Team Review" label so someone from your team can take a look. 👍

Thanks @jeherve! I think @sergeymitr will look into it - ref p1642030666001300/1639517545.011100-slack-CPBTESZMG

**Comment by sergeymitr** [comment]
Hi @htdat.

I'm worried that the `notice` class name is too generic, and other plugins and themes might unintentionally break the styling.
It already affects the styling (see screenshot). We can fix that we know of, but we can't anticipate everything third-party plugins or themes will do.

Is there any other class name we could use?

The screenshot.
The grey outer border should not be there, it's added by `.notice`.
Screen Shot 2022-01-13 at 11 38 55 AM

**Comment by htdat** [comment]
> I'm worried that the notice class name is too generic, and other plugins and themes might unintentionally break the styling.
We can fix that we know of, but we can't anticipate everything third-party plugins or themes will do.

I don't think third-party plugins or themes usually customize this class name as the current stlyes provided by WP core [src](https://github.com/WordPress/WordPress/blob/b4de570d3555cce5d386f2f10d1d49d2444f3a77/wp-admin/css/common.css#L1376) is quite enough.

> It already affects the styling (see screenshot). 

Can you try rebuilding the CSS file? My CSS change here is to override these effects.

> Is there any other class name we could use?

Only other options are `updated` and `error`, which is not as good as `notice` IMO. 

Ref: https://github.com/WordPress/WordPress/blob/master/wp-admin/js/common.js#L1083 
I found out this from the ref here https://github.com/woocommerce/woocommerce-admin/blob/0b68c1d7c0c5025789071022efa129126da1ded5/src/Loader.php#L724-L739)

**Review by sergeymitr — APPROVED** [review]
Hi @htdat.

> Can you try rebuilding the CSS file? My CSS change here is to override these effects.

I tried again and everything looked fine. It does look like something went wrong with rebuilding CSS last time I tested.

🚢 👍

**Review by jeherve — APPROVED** [review]

**Pull request merged**